core/products/perpetual.go

```diff
@@ -46,6 +46,13 @@ const (
 	dataPointSourceInternal dataPointSource = eventspb.FundingPeriodDataPoint_SOURCE_INTERNAL
 )
 
+type fundingData struct {
+	fundingPayment *num.Int
+	fundingRate    *num.Decimal
```

Comment:
Why is fundingRate a pointer? usally we neved use them as pointer.
Also I would have made the fundingPayment a Uint, it sound weird to me to have a negative amount to be paid, I guess you can figure where the payment is going basd on the fundingRate being negative or positive?

Reply:
althought I realise it's already un Int everywhere, I'm just having this thoughts about it now.

Reply:
I don't know why its a pointer, I'm made it not one.

The funding payment is signed because by definition it is the amount being paid and the direction, and the funding-rate is just the funding payment scaled so I think it would be more weird to not have them both signed:
```
fundingPayment = (f_twap - s_twap)
fundingRate = funding_payment /  s_twap
```